### PR TITLE
Add tutorial for Kubernetes launch cluster

### LINKED ISSUE #3743: Write tutorial for kubernetes launch cluster
The existing kubernetes tutorial assumes a preexisting cluster, Toil can create kubernetes clusters, we just don't tell anyone how.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-987)
┆friendlyId: TOIL-987

### PR DESCRIPTION
**Pull request by w-gao**
Fixes #3743. Update documentation to include a small tutorial on how to launch a Kubernetes cluster with Toil.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Add tutorial for Kubernetes launch cluster

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: docs/running/cloud/amazon.rst [hunk 1]
```diff
@@ -361,6 +361,120 @@ non-preemptible nodes of that type were specified in ``--nodeTypes``.
...
+Remember to replace ``<cluster-name>`` with your actual cluster name, and feel
+free to use your own cluster configuration and/or workflow files. For more
+information on this step, visit :ref:`StaticProvisioning`.
+
+Now that we are inside the cluster, a Kubernetes environment should have already
+been configured and running. To verify this, simply run: ::
+
+    $ kubectl get nodes
+
+You should see a ``master`` node with the ``Ready`` status. Depending on the
```
Comment (adamnovak) [inline docs/running/cloud/amazon.rst:421]:
```suggestion
You should see a leader node with the ``Ready`` status. Depending on the
```

Is our Kubernetes version still calling them `master` nodes? I thought Kubernetes switched to "control plane", while Toil mostly says generic "leader" for all schedulers.

File: docs/running/cloud/amazon.rst [hunk 2]
```diff
@@ -361,6 +361,120 @@ non-preemptible nodes of that type were specified in ``--nodeTypes``.
...
+number of worker nodes you set to create upfront, you should also see them
+displayed here.
+
+Additionally, you can also verify that the metrics server is running: ::
+
+    $ kubectl get --raw "/apis/metrics.k8s.io/v1beta1/nodes"
+
+If there is a JSON response and you are not seeing any errors, that means the
```
Comment (adamnovak) [inline docs/running/cloud/amazon.rst:429]: Maybe there could be an example JSON response? Not everyone knows JSON on sight.

File: docs/running/cloud/amazon.rst [hunk 3]
```diff
@@ -361,6 +361,120 @@ non-preemptible nodes of that type were specified in ``--nodeTypes``.
...
+
+Now we can run the workflow: ::
+
+    $ python sort.py \
+            --provisioner aws
+            --batchSystem kubernetes \
+            aws:<region>:<job-store-name>
+
+Make sure to replace ``<region>`` and ``<job-store-name>``. It is **required**
+to use an AWS job store when using the Kubernetes batch system.
```
Comment (adamnovak) [inline docs/running/cloud/amazon.rst:447]:
```suggestion
to use a cloud-accessible job store like AWS or Google when using the Kubernetes batch
system.
```

File: docs/running/cloud/kubernetes.rst [hunk 4]
```diff
@@ -17,6 +17,8 @@ Preparing your Kubernetes environment
 
    To run Toil workflows on Kubernetes, you need to have a Kubernetes cluster set up. This will not be covered here, but there are many options available, and which one you choose will depend on which cloud ecosystem if any you use already, and on pricing. If you are just following along with the documentation, use ``minikube`` on your local machine.
 
+   Alternatively, if you are using the :ref:`Toil provisioner <installProvisioner>`, Toil can set up a Kubernetes cluster for you. Follow :ref:`this <ProvisioningWithKubernetes>` guide to get started with a Toil-managed Kubernetes cluster on AWS.
```
Comment (adamnovak) [inline docs/running/cloud/kubernetes.rst:20]:
```suggestion
   Alternatively, Toil can set up a Kubernetes cluster for you with the :ref:`Toil provisioner <installProvisioner>`. Follow :ref:`this <ProvisioningWithKubernetes>` guide to get started with a Toil-managed Kubernetes cluster on AWS.
```

I don't think using or not using the provisioner is like an overall state that needs to be coped with; they can go get the provisioner setup if they need it.

File: docs/running/cloud/amazon.rst [hunk 5]
```diff
@@ -361,6 +361,120 @@ non-preemptible nodes of that type were specified in ``--nodeTypes``.
...
+
+
+Below is a tutorial on how to launch a Toil job on this newly created cluster.
+As a demostration, we will use :download:`sort.py <../../../src/toil/test/sort/sort.py>`
+again, but run it on a Toil cluster with Kubernetes. First, download this file
+and put it to the current working directory.
+
+We then need to copy over the workflow file and SSH into the cluster: ::
+
+    (venv) $ toil rsync-cluster -z us-west-2a <cluster-name> tutorial_multiplejobs3.py :/root
```
Comment (adamnovak) [inline docs/running/cloud/amazon.rst:408]:
```suggestion
    (venv) $ toil rsync-cluster -z us-west-2a <cluster-name> sort.py :/root
```
This should be the same file as elsewhere in the tutorial.

**Review by adamnovak — CHANGES_REQUESTED** [review]
This looks pretty good, but I have some little corrections I think we need.

File: docs/running/cloud/kubernetes.rst (hunk 4 above)
Comment (w-gao) [reply docs/running/cloud/kubernetes.rst]: Makes sense!

File: docs/running/cloud/amazon.rst (hunk 1 above)
Comment (w-gao) [reply docs/running/cloud/amazon.rst]: Yeah,

```
root@ip-172-31-26-237:/# kubectl get nodes
NAME                                          STATUS   ROLES    AGE   VERSION
ip-172-31-26-237.us-west-2.compute.internal   Ready    master   37h   v1.19.3
ip-172-31-5-230.us-west-2.compute.internal    Ready    <none>   37h   v1.19.3
```

It looks like Kubernetes changed it in [v1.20](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md#urgent-upgrade-notes) but we are using [v1.19.3](https://github.com/DataBiosphere/toil/blob/releases/5.9.x/src/toil/provisioners/abstractProvisioner.py#L840). Changing it to "leader" makes sense though.

**Review by adamnovak — APPROVED** [review]

**Pull request merged**